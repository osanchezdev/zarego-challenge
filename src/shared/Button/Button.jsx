import React from 'react';
import { oneOf, bool, func, oneOfType, arrayOf, node } from 'prop-types';
import styles from './Button.module.css';

const Button = ({ children, disabled, type, variant, onClick }) => (
  <div className={styles.button__wrapper}>
    <button
      disabled={disabled}
      type={type}
      className={`${styles.button_default} ${
        variant === 'primary' ? styles.button_primary : styles.button_secondary
      }`}
      onClick={onClick}
    >
      {children}
    </button>
  </div>
);

Button.propTypes = {
  children: oneOfType([arrayOf(node), node]).isRequired,
  disabled: bool,
  type: oneOf(['button', 'submit']),
  variant: oneOf(['primary', 'secondary']),
  onClick: func,
};

Button.defaultProps = {
  disabled: false,
  type: 'button',
  variant: 'primary',
  onClick: null,
};

export default Button;
