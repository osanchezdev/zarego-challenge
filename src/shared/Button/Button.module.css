.button__wrapper {
}

.button_default {
  padding: 7px 10px;
  border: none;
  min-width: 90px;
  text-align: center;
  cursor: pointer;
}
.button_default:disabled {
  cursor: default;
}

.button_primary,
.button_primary:focus,
.button_primary:active,
.button_primary:active:focus,
.button_secondary,
.button_secondary:focus,
.button_secondary:active,
.button_secondary:active:focus {
  outline: none;
}

.button_primary {
  background: #666;
  color: #eee;
  letter-spacing: 1px;

  transition: background 300ms ease;
}
.button_primary:hover {
  background: #444;
}

.button_secondary {
  background: #eee;
  color: #666;
  letter-spacing: 1px;

  transition: background 300ms ease;
}
