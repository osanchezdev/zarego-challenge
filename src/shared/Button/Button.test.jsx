import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Button from './Button';

describe('Button component snapshot', () => {
  it('should render component', () => {
    const tree = renderer
      .create(
        <Button>
          <span>Test</span>
        </Button>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Button component funtional tests', () => {
  it('should render children', () => {
    render(
      <Button>
        <span>Test</span>
      </Button>
    );
    const children = screen.getByText('Test');
    expect(children).toBeInTheDocument();
  });

  it('should be disabled', () => {
    render(
      <Button disabled>
        <span>Test</span>
      </Button>
    );
    const button = screen.getByRole('button');
    expect(button).toBeDisabled();
  });

  it('should fire handleClick function', () => {
    const handleClick = jest.fn();
    render(
      <Button onClick={handleClick}>
        <span>Test</span>
      </Button>
    );
    const button = screen.getByRole('button');
    fireEvent.click(button);
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
