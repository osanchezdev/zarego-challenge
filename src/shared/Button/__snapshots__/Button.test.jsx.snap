// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component snapshot should render component 1`] = `
<div
  className="button__wrapper"
>
  <button
    className="button_default button_primary"
    disabled={false}
    onClick={null}
    type="button"
  >
    <span>
      Test
    </span>
  </button>
</div>
`;
